### ISSUE TITLE
Issue with the "make clean" target in the top-level directory

### ISSUE THREAD

**Issue by rohgarg** [dmtcp/dmtcp#585]
A user reports that:

>  make clean contains a line which "rm -rf"s the DMTCP_TMPDIR. So when I
> ran "make clean" as root.....the system-wide /tmp directory got removed and
> a lot of things broke!

We need to investigate and fix this.

**Comment by heroxbd** [comment]
How about using `rm -rf ${DMTCP_TMPDIR}/dmtcp*` instead?

**Comment by rohgarg** [comment]
> How about using `rm -rf ${DMTCP_TMPDIR}/dmtcp*` instead?

The `dmtcp*` in `${DMTCP_TMPDIR}/dmtcp*` is redundant. `${DMTCP_TMPDIR}` is supposed to refer to `$TMPDIR/dmtcp-*` or `/tmp/dmtcp-*`, so I'm not sure why this issue was seen.

**Comment by heroxbd** [comment]
The story could be the user took `DMTCP_TMPDIR` by its face and thought "ah-ha it is a tmpdir I should override".  Then he executed `make clean DMTCP_TMPDIR=$HOME/directory-that-I-dump-random-things`.

> ${DMTCP_TMPDIR} is supposed to refer to $TMPDIR/dmtcp-*

This definition is misleading.  The variable name suggested a directory instead of a glob pattern.

**Comment by jo11111** [comment]
I am the user that had this problem. I cant remember why, but I set DMTCP_TMPDIR in /etc/profile on all my machines to /tmp , I'm guessing I did this just to be clear that that is where the temp files should go.
So the rm -rf ${DMTCP_TMPDIR} in the Makefile erases the entire /tmp directory.
There is no guarantee that the user will set a temp directory that is exclusive to dmtcp, so I think "rm -rf" is a bit strong. If dmtcp only makes files called ${DMTCP_TMPDIR}/dmtcp* then why not onlt remove those?

**Comment by rohgarg** [comment]
Thanks for reporting this, @jo11111. I think I understand the problem now. Here's what the Makefile says:

```Makefile
clean: tidy
        ...
        - if test -z "$$DMTCP_TMPDIR"; then \
           if test -z "$$TMPDIR"; then \
             DMTCP_TMPDIR=/tmp/dmtcp-$$USER@`/bin/hostname`; \
           else \
             DMTCP_TMPDIR=$$TMPDIR/dmtcp-$$USER@`/bin/hostname`; \
           fi; \
         fi; \
         rm -rf $$DMTCP_TMPDIR
```
Note that the Makefile tests for whether the `$DMTCP_TMPDIR` environment variable is set, and if it is, then it just goes ahead and does an `rm -rf` on directory pointed to by the environment variable. In your case, this directory happened to be the `/tmp` directory. The Makefile skipped past the inner if case which would have set the `$DMTCP_TMPDIR` to `$TMPDIR/dmtcp-*` or `/tmp/dmtcp-*`.

I'm not sure what's the best way to resolve this though. For example, a user could have set the `$DMTCP_TMPDIR` the root directory (`/`), which would also result in breaking a lot of things. We should *definitely* improve the documentation and clarify the intent of this environment variable. Do you have other suggestions/thoughts?

**Comment by heroxbd** [comment]
Yes, I think documenting `DMTCP_TMPDIR` is not a directory would help.
